### PR TITLE
Fix auth urls

### PR DESCRIPTION
**Pull request by jaclarke**

### CODE REVIEW

**Comment by jaclarke** [comment]
> Ahh, nice catch. Without the trailing slash, the new "path" replaces the last path segment, yeah?

Yes, MiroslavPetrik caught it: https://github.com/edgedb/edgedb-js/pull/750#discussion_r1386685436. I think I broke it in the refactor to use `URL`s.

**Comment by jaclarke** [comment]
I'll add some tests for this once I figure out how to mock smtp.

File: packages/auth-core/src/core.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export class Auth {
     const [host, port] = connectConfig.address;
     const baseUrl = `${
       connectConfig.tlsSecurity === "insecure" ? "http" : "https"
-    }://${host}:${port}/db/${connectConfig.database}/ext/auth`;
+    }://${host}:${port}/db/${connectConfig.database}/ext/auth/`;
```
Comment (MiroslavPetrik) [inline packages/auth-core/src/core.ts:34]: here I found one more problem, Next does not want to connect to host of `localhost`, so I have

    const hostFix = host === "localhost" ? "127.0.0.1" : host;


```suggestion
    }://${hostFix }:${port}/db/${connectConfig.database}/ext/auth/`;
```

Comment (MiroslavPetrik) [reply packages/auth-core/src/core.ts:34]: image

original error,
found the solution in 
https://stackoverflow.com/questions/74165121/next-js-fetch-request-gives-error-typeerror-fetch-failed

Not sure if its not only local, my node version is

$ node -v
v18.18.2

Comment (scotttrinh) [reply packages/auth-core/src/core.ts:34]: I think it's related to this issue: https://github.com/nodejs/node/pull/44731

I believe this is fixed in Node v20 (see https://github.com/nodejs/node/pull/44731), but maybe there is something we can do to opt-into the same behavior in our codebase here. I'll look into that.

Comment (MiroslavPetrik) [reply packages/auth-core/src/core.ts:34]: Just tested with node v20.9.0, and it still fails

Comment (scotttrinh) [reply packages/auth-core/src/core.ts:34]: What is your operating system? If Node v20 is still failing in the same way, it seems like the DNS resolver is not returning `127.0.0.1` for `localhost` lookup which is definitely odd.

Comment (MiroslavPetrik) [reply packages/auth-core/src/core.ts:34]: I'm on Windows 11. Solved it with (for now)
```
import dns from "node:dns";

dns.setDefaultResultOrder("ipv4first");
```
it also impacted the default edgedb client connection. You are right with the DNS resolver, it prioritizes the ipv6, as in the screenshot of the error we see `::1`. 

I don't know where to change the priority in the OS.

Comment (scotttrinh) [reply packages/auth-core/src/core.ts:34]: @MiroslavPetrik perhaps this comment might help: https://github.com/nodejs/node/issues/40537#issuecomment-1706257550

Comment (MiroslavPetrik) [reply packages/auth-core/src/core.ts:34]: Thanks, will save that as alternative option for developing on windows 11.

Still it seems temporary, just noticed that even the hosts file no longer uses localhost mapping (they are commented out) as that somehow interferes with ipv6 adoption, so maybe the edgedb could listen on ipv6 as well? 
I don't know much about networking. Just expanding on what I just read

Comment (jaclarke) [reply packages/auth-core/src/core.ts:34]: Yes, it's possible to make edgedb listen on ipv6 as well. There was a similar issue previously with Deno resolving localhost to the ipv6 `::1` address: https://github.com/edgedb/edgedb-js/issues/376#issuecomment-1173840632

**Pull request merged**